src/isValidInfoFile.js

```diff
@@ -0,0 +1,120 @@
+const { readInfoFile } = require("./utils");
+
+// helper function to determine the validity of the provided infofile type
+function isValidType(type) {
+  switch (type) {
+    case "Vehicle":
+      return true;
+    case "TestRun":
+      return true;
+    case "Model":
+      return true;
+    case "Road":
+      return true;
+    default:
+      return false;
+  }
+}
+
+// a function to determine whether a file is an infofile or not
+function isValidInfoFile({ file, type }) {
+  // check that the file has been provided
+  if (!file) {
+    throw new Error("file is required");
+  }
+
+  // check that type has been provided
+  if (!type) {
+    throw new Error("type is required");
+  }
+
+  if (!isValidType(type)) {
+    throw new Error("type is invalid");
+  }
+
+  // read the info file
+  const infofile = readInfoFile(file);
+
+  try {
+    const fileIdent = infofile.getString("FileIdent");
+
+    // looks for the appropriate condition to validate the infofile based on the provided type
+    switch (type) {
+      // regex match for possible vehicle types in the FileIdent property of the infofile
+      case "Vehicle":
+        if (
+          fileIdent.match(/^CarMaker-Car\s\d{2}$/) ||
+          fileIdent.match(/^CarMaker-Truck\s\d{2}$/) ||
+          fileIdent.match(/^CarMaker-Motorcycle\s\d{2}$/)
+        ) {
+          // delete the infofile handle
+          infofile.delete();
+          return true;
+        } else {
+          // delete the infofile handle
+          infofile.delete();
+          return false;
+        }
```

Comment:
Generally speaking, if else statements already operate on a boolean, so there is no need to return booleans from within. Instead just return the statement. Same applies to the rest of the switch cases.

```suggestion
        return fileIdent.match(/^CarMaker-Car\s\d{2}$/) ||
          fileIdent.match(/^CarMaker-Truck\s\d{2}$/) ||
          fileIdent.match(/^CarMaker-Motorcycle\s\d{2}$/)
```

Granted here you are also handling infofile.delete() which I'll comment on seperately.

Reply:
The if-else statements are operating on vectors (.match returns that), so this solution cannot be used here. However, with the elimination of all the infofile.delete-s, I can use the ? : operator now, which will make the code a bit more compact

Reply:
Makes sense